loopy/transform/iname.py

```diff
@@ -1126,26 +1128,56 @@ def has_schedulable_iname_nesting(kernel):
 # {{{ rename_inames
 
 @for_each_kernel
-def rename_iname(kernel, old_iname, new_iname, existing_ok=False, within=None):
+def rename_inames(kernel, old_inames, new_iname, existing_ok=False, within=None):
     """
+    :arg old_inames: A collection of inames that must be renamed to **new_iname**.
     :arg within: a stack match as understood by
         :func:`loopy.match.parse_stack_match`.
     :arg existing_ok: execute even if *new_iname* already exists
     """
+    from collections.abc import Collection
+    if (isinstance(old_inames, str)
+            or not isinstance(old_inames, Collection)):
+        raise LoopyError("'old_inames' must be a collection of strings, "
+                         f"got '{type(old_inames)}'.")
+
+    if any((len(insn.within_inames & frozenset(old_inames)) > 1)
+           for insn in kernel.instructions):
+        raise LoopyError("old_inames contains nested inames"
+                         " -- renaming is illegal.")
+
+    # sort to have deterministic implementation.
+    old_inames = sorted(old_inames)
 
     var_name_gen = kernel.get_var_name_generator()
 
     # FIXME: Distinguish existing iname vs. existing other variable
     does_exist = var_name_gen.is_name_conflicting(new_iname)
```

Comment:
`is_name_conflicting` doesn't entail that the name already exists. It's some custom weirdness, see here:

https://github.com/inducer/loopy/blob/9e69cf4b2dedad95e5f52ac5d7a8fbe1a08a8c80/loopy/kernel/__init__.py#L71-L89

This should probably just be `new_iname in kernel.all_inames()`. (I know it's not your fault, but I think it's worth fixing now.)

Reply:
Yep, done. Also, we raise an error now if 'new_iname' is any other variable name in the kernel.